pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -222,24 +241,19 @@ func (c *DeploymentConfigController) findDetails(config *deployapi.DeploymentCon
 		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
 			if !errors.IsNotFound(err) {
 				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
-				return details, existingDeployments, latestDeploymentExists, nil
+				return details, nil
 			}
 			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
 			// (a build configuration output points to it so it's going to be populated at some point in the
 			// future)
 			details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.", istag, istag)
-			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-			if err != nil {
-				glog.V(2).Infof("Error while trying to list build configs: %v", err)
-				return details, existingDeployments, latestDeploymentExists, nil
-			}
-
-			for _, bc := range bcList.Items {
+			for _, obj := range c.buildConfigs.List() {
```

Comment:
This code guarantees the updates only occur if a.) the status has never been updated before or b.) the status is more than a minute old

I also just realized that we can't filter buildConfigs on namespace using a store because the store is constructed for use by the controller generally during init, not per handle. The choices seem to be a.) work against a (possibly filtered) store of buildConfigs across all namespaces or b.) work with a fresh (possibly filtered) list of buildConfigs in the namespace of the current config we're handling. Option a seems most flexible to me in terms of optimization potential.